File: 백준/Silver/4948. 베르트랑 공준/베르트랑 공준.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
	static boolean[] check = new boolean[123456*2+1];
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		prime_check();
		
		while(true) {
			int n = Integer.parseInt(br.readLine());
			if(n==0) break;
			int count=0;
			for(int i = n+1; i <=n*2 ; i++) 
				if(!check[i]) count++;
			
			System.out.println(count);
			
		}
	
	}
	
	static void prime_check() {
		
		check[0]=check[1]=true;
		
		for(int i=2; i<Math.sqrt(check.length);i++) {
			if(check[i]==true)
				continue;
			for(int j=i * i; j<check.length;j+=i)
				check[j]=true;
		}
	}
}

File: 백준/Silver/4948. 베르트랑 공준/베르트랑 공준.js
const readline = require('readline').createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];

readline.on('line', function(line) {
  input.push(line)
}).on('close', function () {
  const ns = input.map(Number)
  const isPrime = sieve()
  let answer = []
  for (const n of ns) {
    if (n !== 0) {
      let count = 0;
      for (let i = n + 1; i <= 2 * n; i++) {
        if (isPrime[i])
          count++
      }
      answer.push(count)
    }
  }

  console.log(answer.join("\n"))
});

function sieve() {
  const MAX_NUM = 123456 * 2
  const isPrime = new Array(MAX_NUM + 1).fill(true)
  const sqrt = Math.sqrt(MAX_NUM)
  isPrime[0] = false
  isPrime[1] = false 
 
  for (let i = 2; i <= sqrt; i++) {
    if (!isPrime[i]) continue
    for (let j = i * i; j <= MAX_NUM; j += i) {
      isPrime[j] = false
    }
  }
  return isPrime
}